bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/handler/Enigma2Handler.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enigma2.handler;
+
+import static org.openhab.binding.enigma2.internal.Enigma2BindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.*;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.enigma2.actions.Enigma2Actions;
+import org.openhab.binding.enigma2.internal.Enigma2Client;
+import org.openhab.binding.enigma2.internal.Enigma2Configuration;
+import org.openhab.binding.enigma2.internal.Enigma2RemoteKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The {@link Enigma2Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Guido Dolfen - Initial contribution
+ */
+@NonNullByDefault
+public class Enigma2Handler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(Enigma2Handler.class);
+    private Optional<Enigma2Configuration> configuration = Optional.empty();
+    private Optional<Enigma2Client> enigma2Client = Optional.empty();
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private LocalDateTime lastAnswerTime = LocalDateTime.now();
+
+    public Enigma2Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = Optional.ofNullable(getConfigAs(Enigma2Configuration.class));
+        configuration.ifPresent(config -> {
```

Comment:
If there is no configuration present shouldn't that put the thing offline?

Reply:
Fixed, because of changing to a default Enigma2Configuration the configuration never clould be null. If it is only the default, there are missing parameters and so the thing would go offline in case of parameter validation